crypto/keys/private.go

```diff
@@ -23,8 +24,16 @@ import (
 	"errors"
 	"fmt"
 	"io/ioutil"
+
+	"github.com/google/trillian"
 )
 
+// Provider handles acquisition of signers for trees.
+type Provider interface {
+	// Signer returns a signer for the given tree.
+	Signer(context.Context, *trillian.Tree) (crypto.Signer, error)
```

Comment:
Yeah, I get that it feels a bit more like a "Get", but in another sense, it's a "New" because it creates a new object that meets the `crypto.Signer` interface. It just fills that object with info that it's fetching from elsewhere.  

It might be clearer for users to use something other than `Signer` though, since it's also the name of an interface and an object already.  No need to reinvent [buffalo buffalo](https://en.wikipedia.org/wiki/Buffalo_buffalo_Buffalo_buffalo_buffalo_buffalo_Buffalo_buffalo) :-)